### PR TITLE
Add support for config payload for AuditConfiguration

### LINKED ISSUE #164: Enhancement: Provide a config object to AuditConfiguration constructor
I think it would be nice to be able to provide a configuration payload to the `AuditConfiguration` constructor instead of creating it and then setting each option one by one.

This would allow for config payloads to be reusable when running a batch of audits.

i.e.

```javascript
var config, auditConfig, audit, report;

config = {
  scope: document.querySelector('.active'),
  showUnsupportedRulesWarning: false,
  maxResults: 1
};

auditConfig = new axs.AuditConfiguration(config);
audit = axs.Audit.run(auditConfig);
report = axs.Audit.createReport(audit);
```

### PR DESCRIPTION
**Pull request by loginx**
* Overwrite `AuditConfiguration` properties defined in the constructor with properties found in payload parameter.
* Add test for new behaviour.
* Add usage example in docs.
* Fixes #164

### CODE REVIEW

File: src/js/Audit.js [hunk 1]
```diff
@@ -22,8 +22,13 @@ goog.provide('axs.AuditConfiguration');
 /**
  * Object to hold configuration for an Audit run.
  * @constructor
+ * @param {?Object=} config Configuration object
  */
-axs.AuditConfiguration = function() {
+axs.AuditConfiguration = function(config) {
```
Comment (ckundo) [inline src/js/Audit.js]: Cool! Me gusta.

**Comment by ckundo** [comment]
This is an exciting change, great work! LGTM

File: src/js/Audit.js [hunk 2]
```diff
@@ -78,6 +83,12 @@ axs.AuditConfiguration = function() {
      */
     this.showUnsupportedRulesWarning = true;
 
+    for (var prop in this) {
+        if ((this.hasOwnProperty(prop)) && (prop in config)) {
+            this[prop] = config[prop];
```
Comment (alice) [inline src/js/Audit.js]: This is cool, but is not going to work for `severity`, `ignoreSelectors` or `ruleConfig`. We'll probably need to special case those.

**Comment by loginx** [comment]
@alice That's correct. That can be done but we have to determine a reasonable format for the data to pass to these methods in the config payload. 

It would probably be a good idea to merge this in first, and open an issue for automating these methods through config as well.

File: src/js/Audit.js [hunk 3]
```diff
@@ -22,8 +22,13 @@ goog.provide('axs.AuditConfiguration');
 /**
  * Object to hold configuration for an Audit run.
  * @constructor
+ * @param {?Object=} config Configuration object
```
Comment (alice) [inline src/js/Audit.js]: Can you add more detailed documentation for this, stating what can and cannot be configured this way?

Comment (loginx) [inline src/js/Audit.js]: Will do.

**Comment by ckundo** [comment]
Looking forward to getting this merged! @loginx do you need an support on this?

**Comment by loginx** [comment]
@ckundo I completely forgot about this. I thought it had been merged long ago. I'll have a look at this tonight as well.

**Comment by alice** [comment]
:+1: go ahead and pull

**Pull request merged**